addons/editor-extra-blocks/userscript.js

```diff
@@ -0,0 +1,219 @@
+          </value>`,
+        }
+      );
+
+      insertAfter(
+        "event_broadcastandwait",
+        true,
+
+        {
+          type: "event_whentouchingobject",
+          innerHTML: `
+        <value name="TOUCHINGOBJECTMENU">
+          <shadow type="event_touchingobjectmenu">
+          </shadow>
+        </value>`,
+        }
+      );
+
+      insertAfter("sensing_loudness", false, { type: "sensing_loud" });
+
+      if (addon.settings.get("counterBlocks")) {
+        insertAfter(
+          "control_delete_this_clone",
+          true,
+
+          { type: "control_get_counter" },
+          { type: "control_incr_counter" },
+          { type: "control_clear_counter" }
+        );
+      }
+
+      if (addon.settings.get("uselessBlocks")) {
+        insertAfter(
+          "motion_sety",
+          true,
+
+          { type: "motion_align_scene" }
+        );
+
+        insertAfter(
+          "motion_sety",
+          true,
+
+          {
+            type: "motion_scroll_right",
+            innerHTML: `
+            <value name="DISTANCE">
+              <shadow type="math_number">
+                  <field name="NUM">10</field>
+              </shadow>
+            </value>`,
+          },
+          {
+            type: "motion_scroll_up",
+            innerHTML: `
+            <value name="DISTANCE">
+              <shadow type="math_number">
+                  <field name="NUM">10</field>
+              </shadow>
+            </value>`,
+          }
+        );
+
+        insertAfter(
+          "motion_direction",
+          true,
+
+          { type: "motion_xscroll" },
+          { type: "motion_yscroll" }
+        );
+
+        insertAfter(
+          "looks_hide",
+          false,
+
+          { type: "looks_hideallsprites" }
+        );
+
+        insertAfter(
+          "looks_setsizeto",
+          true,
+
+          {
+            type: "looks_setstretchto",
+            innerHTML: `
+            <value name="STRETCH">
+              <shadow type="math_number">
+                  <field name="NUM">100</field>
+              </shadow>
+            </value>`,
+          },
+          {
+            type: "looks_changestretchby",
+            innerHTML: `
+            <value name="CHANGE">
+              <shadow type="math_number">
+                  <field name="NUM">10</field>
+              </shadow>
+            </value>`,
+          }
+        );
+
+        insertAfter(
+          "control_for_each",
+          false,
+
+          { type: "control_all_at_once" }
+        );
+
+        insertAfter(
+          "sensing_username",
+          false,
+
+          { type: "sensing_userid" }
+        );
+      }
+
+      addon.tab.redux.dispatch({
+        type: UPDATE_TOOLBOX_ACTION,
+        toolboxXML: xmlSerializer.serializeToString(toolboxXML),
+        saExtraBlocks: true,
+      });
+    }
+  };
+
+  const updateToolbox = () => {
+    if (vm.editingTarget) {
+      vm.emitWorkspaceUpdate();
```

Comment:
This method clears the undo stack, which isn't an ideal side-effect of changon addon settings or enabling/disabling the addon.

Reply:
Related: https://github.com/ScratchAddons/ScratchAddons/pull/6727#issuecomment-1752023575

Reply:
[`updateToolboxXML()`](https://github.com/ScratchAddons/ScratchAddons/blob/9773b46/libraries/common/cs/update-toolbox-xml.js) now exists.